src/dearpygui_commands.h

```diff
@@ -11,6 +11,11 @@
 #include "stb_image_write.h"
 #include "mvProfiler.h"
 #include "mvUtilities.h"
+#include <string.h>
+
+#ifndef strnicmp
```

Comment:
Shame on me, I believed `strnicmp` to be a part of C++ standard, which it is not! It didn't occur to me that even now in 2020's, C++ might still not have a standard function to perform case-insensitive comparisons.

`strnicmp` does not exist in Microsoft CRT; the following MSDN page says it's named `_strnicmp` instead:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/strnicmp-wcsnicmp-mbsnicmp-strnicmp-l-wcsnicmp-l-mbsnicmp-l

As a result, the `#ifndef` doesn't really work, and the build fails on Windows.

What's worse, on the version of Visual Studio I have, `_strnicmp` is a function, not a macro. Therefore `#ifdef` won't be able to detect whether it exists. I wonder if we can revert the condition and check for `strncasecmp` instead, but it's a hack anyway (not sure if any standard guarantees `strncasecmp` to be a macro). Alternatively, we could choose the function based on `#ifdef _WIN32`.

Yet another alternative is to convert the file name to lowercase beforehand (right there in `save_image`) and compare it using `strncmp`. This latter sounds trivial but it's not quite - conversion to lowercase will look like this:

```cpp
	std::string file_str(file);
	std::transform(
		file_str.begin(), file_str.end(), file_str.begin(), 
		[](unsigned char c){ return tolower(c); }
	);
	file = file_str.c_str();
```

Still I'm inclined to use the latter solution because personally I prefer as little `#ifdef` magic as possible. My own local version does just that - convert to lowercase just before that sequence of `if (str_ends_with...` statements, and use plain `strncmp` in `str_ends_with`.

P.S. Sorry for the confusion I caused by starting this case-insensitive-comparison discussion.

Reply:
No problem!
I chose to go for the `#ifdef _WIN32` because it seemed to me the best compromise between simplicity and reliability. This solution is really easy to mantain and, eventually, replace with this function (if there is ever a need).